pallets/posts/src/functions.rs

```diff
@@ -318,6 +318,77 @@ impl<T: Trait> Module<T> {
         Self::share_post(creator.clone(), original_post, new_post_id)
     }
 
+    fn mutate_posts_count_on_space<F: FnMut(&mut u32) + Copy> (
+        space_id: SpaceId,
+        post: &Post<T>,
+        mut f: F
+    ) -> DispatchResult {
+        Spaces::<T>::mutate_space_by_id(space_id, |space: &mut Space<T>| {
+            if post.hidden {
+                f(&mut space.hidden_posts_count);
+                f(&mut space.posts_count);
+            } else {
+                f(&mut space.posts_count);
+            }
+        }).map(|_| ())
+    }
+
+    pub(crate) fn move_post_to_space(
+        editor: T::AccountId,
+        post: &mut Post<T>,
+        new_space_id: SpaceId
+    ) -> DispatchResult {
+        let old_space_id = post.get_space_id()?;
+        let new_space = Spaces::<T>::require_space(new_space_id)?;
+
+        Spaces::ensure_account_has_space_permission(
+            editor,
+            &new_space,
+            SpacePermission::CreatePosts,
+            Error::<T>::NoPermissionToCreatePosts.into()
+        )?;
+
+        // TODO check whether post and its content are not blocked within a new space
```

Comment:
There should be a way to set up checks here without implementing pallet-moderation into the Runtime